packages/data/src/types.ts

```diff
@@ -0,0 +1,8 @@
+// Type for the basic selectors built into @wordpress/data, note these
+// types define the interface for the public selectors, so state is not an
+// argument.
+export type WPDataSelectors = {
+	hasStartedResolution: ( selector: string, args?: string[] ) => boolean;
```

Comment:
This doesn't cover everything WPDataSelectors has, I just started with the most commonly used selectors that we use all the time.

Reply:
Would it be worth to add this to typescript declaration file? I did add a `~/typings` folder for the `d.ts` files.
We could create one for @wordpress/data for now:

`declare module "@wordpress/data" { ... }` or create a folder within `@wordpress-data` with an `index.d.ts`. 
This doesn't have to be part of this PR, but just a thought as we iterate, and wait for @wordpress/data to be written in TS.

Reply:
I was hesitant to create one for wordpress__data. It does have its own types but they just aren't very useful to us right now. This is used to compose a type of these methods and those on the wc stores so its not really intended to provide type declarations for wp data.

Reply:
No worries, I thought I mention it, all good with me.

Reply:
@louwie17 I appreciate it for sure. I still feel not great about ignoring `@types/wordpress__data` Maybe we can look at whether we could provide some upstream help to them to make the types better going forward. Its something I'm definitely thinking about.

Reply:
I've declared modules for the 2 @wordpress packages and I've added the type defs for lodash.